net2cog/netcdf_convert.py

```diff
@@ -62,51 +62,62 @@ def _write_cogtiff(
         example :/home/dockeruser/converter/podaac/netcdf_converter/temp/
             netcdf_converter/
             RSS_smap_SSS_L3_8day_running_2020_037_FNL_v04.0_test
-    nc_xarray : xarray.Dataset
-        xarray dataset loaded from NetCDF file
-    variable_name: str
-        Name of the variable within the file to convert.
+    nc_xarray : xarray.DataTree
+        xarray DataTree loaded from NetCDF file. This represents the whole
+        file.
+    variable_path: str
+        Full of the variable within the file to convert.
     logger : logging.Logger
         Python Logger object for emitting log messages.
 
     Notes
     -----
     - Assumption that 0 is always on the prime meridian/equator.
-    - The output name for converted GeoTIFFs is `<variable name>.tif`, with any
+    - The output name for converted GeoTIFFs is `<variable path>.tif`, with any
       slashes replaced with underscores.
     """
 
-    logger.debug("NetCDF Var: %s", variable_name)
+    logger.debug("NetCDF Var: %s", variable_path)
 
-    if variable_name in EXCLUDE_VARS:
-        logger.debug(f"Variable {variable_name} is excluded. Will not produce COG")
+    if variable_path in EXCLUDE_VARS:
+        logger.debug(f"Variable {variable_path} is excluded. Will not produce COG")
         return None
 
-    output_basename = f'{variable_name}.tif'.replace('/', '_')
+    output_basename = f'{variable_path}.tif'.lstrip('/').replace('/', '_')
     output_file_name = path_join(output_directory, output_basename)
 
     with TemporaryDirectory() as tempdir:
         temp_file_name = path_join(tempdir, output_basename)
 
         try:
-            nc_xarray[variable_name].rio.to_raster(temp_file_name)
+            if not has_spatial_dimensions(nc_xarray[variable_path]):
+                # The variable being processed does not have spatial dimensions:
+                raise Net2CogError(
+                    variable_path,
+                    f'{variable_path} does not have spatial dimensions such as'
+                    'lat / lon or x / y'
+                )
+            nc_xarray[variable_path].rio.to_raster(temp_file_name)
         except KeyError as error:
-            # Occurs when trying to locate a variable that is not in the Dataset
-            raise Net2CogError(variable_name, error) from error
+            # Occurs when trying to locate a variable that is not in the DataTree
+            raise Net2CogError(
+                variable_path,
+                f"No variable named '{variable_path}'."
```

Comment:
I explicitly specified the error message here, as `xarray.DataTree` has a different error message:

```
KeyError: 'Could not find node at <variable path>'
```

I thought that raw error was perhaps less intuitive than the older message from the `Dataset`, so wanted to retain the older messaging.